android/build-kaldi.sh

```diff
@@ -45,13 +46,19 @@ mkdir -p $WORKDIR_ARM64/local/lib $WORKDIR_ARM32/local/lib $WORKDIR_X86_64/local
 cd build
 git clone https://github.com/simonlynen/android_libs
 cd android_libs/lapack
-sed -i 's/APP_STL := gnustl_static/APP_STL := c++_static/g' jni/Application.mk && \
-sed -i 's/android-10/android-21/g' project.properties && \
-sed -i 's/APP_ABI := armeabi armeabi-v7a/APP_ABI := armeabi-v7a arm64-v8a x86_64/g' jni/Application.mk && \
-sed -i 's/LOCAL_MODULE:= testlapack/#LOCAL_MODULE:= testlapack/g' jni/Android.mk && \
-sed -i 's/LOCAL_SRC_FILES:= testclapack.cpp/#LOCAL_SRC_FILES:= testclapack.cpp/g' jni/Android.mk && \
-sed -i 's/LOCAL_STATIC_LIBRARIES := lapack/#LOCAL_STATIC_LIBRARIES := lapack/g' jni/Android.mk && \
-sed -i 's/include $(BUILD_SHARED_LIBRARY)/#include $(BUILD_SHARED_LIBRARY)/g' jni/Android.mk && \
+sedCmd=""
+case "${unameOS}" in
+    linux*)     sedCmd=sed;;
+    darwin*)    sedCmd=gsed;;   
+    *)          sedCmd=sed
+esac
+$sedCmd -i 's/APP_STL := gnustl_static/APP_STL := c++_static/g' jni/Application.mk && \
+$sedCmd -i 's/android-10/android-21/g' project.properties && \
+$sedCmd -i 's/APP_ABI := armeabi armeabi-v7a/APP_ABI := armeabi-v7a arm64-v8a x86_64/g' jni/Application.mk && \
+$sedCmd -i 's/LOCAL_MODULE:= testlapack/#LOCAL_MODULE:= testlapack/g' jni/Android.mk && \
+$sedCmd -i 's/LOCAL_SRC_FILES:= testclapack.cpp/#LOCAL_SRC_FILES:= testclapack.cpp/g' jni/Android.mk && \
+$sedCmd -i 's/LOCAL_STATIC_LIBRARIES := lapack/#LOCAL_STATIC_LIBRARIES := lapack/g' jni/Android.mk && \
+$sedCmd -i 's/include $(BUILD_SHARED_LIBRARY)/#include $(BUILD_SHARED_LIBRARY)/g' jni/Android.mk && \
```

Comment:
I think, it will be better to replace `sed -i ` by `sed -i.bak -e `.

Reply:
Thanks for your feedback, I have updated the script.